File: solutions/lua/eliuds-eggs/1/eliuds-eggs.lua
local EliudsEggs = {}

function EliudsEggs.egg_count(number)
	local count = 0
	while number > 0 do
		count = count + (((number % 2) == 1) and 1 or 0)
		number = (number - (number % 2)) / 2
	end
	return count
end

return EliudsEggs

File: solutions/c/eliuds-eggs/1/eliuds_eggs.c
#include "eliuds_eggs.h"

unsigned int egg_count(unsigned int number) {
	int count = 0;
	while (number > 0) {
		count += (((number % 2) == 1) ? 1 : 0);
		number = (number - (number % 2)) / 2;
	}
	return count;
}
